src/lib.rs

```diff
@@ -40,6 +40,7 @@
 //! the client helper as `Client` in the protocol module you're working with (e.g.,
 //! [`pipeline::Client`]), and the server helper as `Server` in the same place.
 //!
+/* TODO: https://github.com/tokio-rs/tokio/pull/3105
```

Comment:
Ouch, losing this example is a bit of a pain. I don't think that's a blocker for this necessarily, though it's worth thinking about what a replacement example might look like. What if we just switched it over to `UnboundedSender` for now (in which case `poll_ready` can simply always return `Ok(())`. It's a little contrived, but would allow the rest of the example to work as-is.

Reply:
Updated to use `unbounded_channel`, thanks for the suggestion!